### PR TITLE
PLNSRVCE-670: Automate the process for upgrading kcp version on the pipeline-service repo

### PR DESCRIPTION
**Pull request by bnallapeta**
- I've also setup a token that is required for authenticating the Github action to create commits/pull requests on the pipeline-service repo.
(token expiration is set to 90 days since we would get rid of actions and start dogfooding our own service)
- Here is an example PR created by this action on my repo during testing - https://github.com/bnallapeta/pipeline-service/pull/25

### CODE REVIEW

File: config/config.yaml [hunk 1]
```diff
@@ -25,3 +25,6 @@ CR_TO_SYNC:
 # Applications to be deployed on the cluster
 APPS:
   - pipeline_service # pipeline_service sets up Pipeline Service on the cluster.
+
+# kcp version
+KCP_VERSION: v0.8.0
```
Comment (Roming22) [inline config/config.yaml:30]: Maybe I would use the following structure:
```
version:
  kcp: v0.8.0
```
We can then add more version/dependencies there in the future if necessary.

File: images/kcp-upgrade/upgrade.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
```
Comment (Roming22) [inline images/kcp-upgrade/upgrade.sh:1]: Not really a "production" image, more related to tooling (as are the test images). I would probably move it out of the way. We might need to think about a slight refactor of the repository.

File: config/config.yaml (hunk 1 above)
Comment (bnallapeta) [reply config/config.yaml:30]: Sounds good. I have made these changes.

File: images/kcp-upgrade/upgrade.sh (hunk 2 above)
Comment (bnallapeta) [reply images/kcp-upgrade/upgrade.sh:1]: Reading this comment, I feel we could actually refactor our repo to a great extent. For eg: ckcp and local can be placed in a new folder 'dev'. We could create 3 folders to store images - dev, test, automation. And place our images accordingly.

I will take this up sometime next week as it is outside the scope of this PR. Let me know what you think.

Comment (Roming22) [reply images/kcp-upgrade/upgrade.sh:1]: Sounds good to me.

File: ckcp/README.md [hunk 3]
```diff
@@ -104,10 +104,11 @@ synctargets                                    workload.kcp.dev/v1alpha1
...
-3. A user can modify the Custom Resources to be synced by the KCP Syncer by modifying [CR_TO_SYNC](./config.yaml).
+2. A user can modify the applications to be deployed on the compute by modifying [APPS](..config/config.yaml).
+3. A user can modify the Custom Resources to be synced by the KCP Syncer by modifying [CR_TO_SYNC](..config/config.yaml).
 4. Onboarding a new component requires creating a new Argo CD application in [argo-apps](../gitops/argocd/argo-apps/) and adding it to [kustomization.yaml](../gitops/argocd/argo-apps/kustomization.yaml).
-5. For testing, users need to modify the git source path and ref of their Argo CD application and [parent Argo CD application](../gitops/argocd/argocd.yaml) which ultimately triggers all the Argo CD applications in [argo-apps](../gitops/argocd/argo-apps/) dir to reflect their own Git repo.
+5. For testing, users need to modify only the git source path and ref of their Argo CD application to reflect their own Git repo.
+6. A user can also choose a different version of kcp than the one running by default by changing the value of [KCP_VERSION](../config/config.yaml).
```
Comment (Roming22) [inline ckcp/README.md:114]: `KCP_VERSION` -> `VERSION.KCP`.
Also what's up with the name being all uppercase, where does that come from?

Comment (bnallapeta) [reply ckcp/README.md:114]: Fixed it. All of the keys are now lowercase.

**Comment by bnallapeta** [comment]
@Roming22 I have addressed the lowercase point as well and all checks seems to pass. Please take a look at the PR and merge if all okay.

**Comment by bnallapeta** [comment]
FYI, the github action is scheduled every week at Sunday 00.00.00.
This time, since we know 0.8.2 is out and we want to upgrade to that, I will manually trigger the action once the PR is merged.

File: ckcp/README.md (hunk 3 above)
Comment (Roming22) [reply ckcp/README.md:114]: I think you forgot to edit `[KCP_VERSION]` to `[version.kcp]`.

Comment (bnallapeta) [reply ckcp/README.md]: Haha, replaced everywhere except the line that you commented on :p
Fixed now.

**Pull request merged**